lib/_http_agent.js

```diff
@@ -127,7 +129,15 @@ function Agent(options) {
     const requests = this.requests[name];
     if (requests && requests.length) {
       const req = requests.shift();
-      setRequestSocket(this, req, socket);
```

Comment:
An open question on the original code: don't we need to call `this.reuseSocket(socket, req);` before `setRequestSocket`?

Reply:
I'm going to ignore this open question and land the PR as it's not related with the fix itself.

Reply:
I agree but as you said this is not related to this PR.

Reply:
I'll submit another PR that addresses the problem.